plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -345,11 +345,11 @@ public IcebergTableHandle getTableHandle(
             throw new TrinoException(NOT_SUPPORTED, "Read table with start version is not supported");
```

Comment:
FWIW the logic is now the same as the logic employed in `DeltaLakeMetadata`:  

https://github.com/trinodb/trino/blob/92e5d111c0dda3d5a9d24a34f975e78724e2e098/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L2506